src/os/bluestore/NVMEDevice.cc

```diff
@@ -845,19 +845,14 @@ void io_complete(void *t, const struct spdk_nvme_cpl *completion)
     task->release_segs(queue);
     // read submitted by AIO
     if(!task->return_code) {
-      if (ctx->priv) {
-	if (!--ctx->num_running) {
-          task->device->aio_callback(task->device->aio_callback_priv, ctx->priv);
-	}
-      } else {
-	ctx->try_aio_wake();
+      assert(ctx->priv != NULL);
```

Comment:
I am not very understand here. Do you mean priv must be passed in. Why?

Reply:
Could you see the if case?  In this case, the priv could not be NULL, if it is NULL, it is definitely a bug, so the original statement is removed.